src/tcp/segment.ml

```diff
@@ -135,10 +136,7 @@ module Rx(Time:Mirage_time.S) = struct
 
   let send_challenge_ack q =
     (* TODO:  rfc5961 ACK Throttling *)
-    (* Is this the correct way trigger an ack? *)
-    if Lwt_mvar.is_empty q.rx_data
-      then Lwt_mvar.put q.rx_data (Some [], Some Sequence.zero)
-      else Lwt.return_unit
+    Lwt_mvar.put q.send_ack Sequence.zero
```

Comment:
I understand you need the new mvar since the rx is dead, but why did you remove the `is_empty` dance ?
I'm more comfortable with the way you did but I'm pondering what made you choose this

Reply:
That's a mistake from my side, it's probably better to keep the `is_empty` check.